rakam-postgresql/src/main/java/org/rakam/postgresql/analysis/FastGenericFunnelQueryExecutor.java

```diff
@@ -96,15 +102,20 @@ public QueryExecution query(String project, List<FunnelStep> steps, Optional<Str
                     filterExp.orElse("true")));
         }
 
-        String dimensions = dimension.map(v -> checkTableColumn(v) + ", ").orElse("");
+        String dimensions = "";
+        if(dimension.isPresent())
+            dimensions = dimension.map(v -> checkTableColumn(v) + ", ").orElse(""); // outer dimensions
```

Comment:
It's usually better practice to have the brackets in one-line if statements because it may lead some strange bugs in the future.
You can install this CheckStyle on your IDEA: https://github.com/airlift/codestyle

Reply:
Complied with project style